### ISSUE TITLE
Update rules to account for private DBQs received within 7 days of claim

### ISSUE THREAD

**Issue by amylai-nava** [department-of-veterans-affairs/abd-vro#1405]
### User Story

As a VRO PO, I want to ensure that claims processed in VRO appropriately handle logic around private DBQs in a way that is consistent with how MAS handles it, so that all claims of a given contention are treated consistently when automated.

### Acceptance Criteria

1. TBD once scope/solution is determined

### Not included in this work

Similar logic applies to 5103 and 4142 notices; these cases will be covered by other tickets (#1489, #1308).

### Notes about work

**[MAS Documentation of APCAS logic for this scenario](https://dvagov.sharepoint.com/:w:/r/sites/vaabdvro/Shared%20Documents/VRO%20-%20Virtual%20Regional%20Office/VRO%20Integrations/MAS%20Integrations/VRO%20Integration%20-%20APCAS%20Standardized%20Automation%20(March%202023).docx?d=w6780247c75fd48f898ad4ff60959fddb&csf=1&web=1&e=JghK6F)**

> 5.9.7. Complete Private DBQ Check
> As an RVSR, I want to make an effective rating decision on all claims that are submitted with condition specific Private DBQ's so that Veterans can avoid unnecessary exams.
> - If a Private DBQ is submitted with a claim on the same day, the claim will be marked RFD.
> - If a Private DBQ is submitted within 7 days of the claim being received, the claim will be marked RFD.
> - If a Private DBQ is submitted for a claim outside of the 7 day period of the claim being received, logic per condition will be followed.
> - If a VA DBQ is found in the eFolder, regardless of day it was submitted, it will have no impact on RFD v Exam logic.
> 
> Private and VA DBQs will no longer require field extractions to be considered complete.
> - DBQs submitted via the mail and OCR'd in the eFolder will always be considered complete and accepted as medical evidence.
> - “[Condition] DBQ" will be displayed in the ‘Schedular Evidence’ Table
> - If greater than 7 days, drives no logic
> - This applies even if the DBQ is incomplete
> 
> This DBQ logic applies across all conditions regardless of the basis of the claim

> 5.9.8. Add VBMS Note for Private DBQ Check
> "_Claim status updated to RFD via automation due to Private DBQ being submitted within 7 days of Date of Claim_"

**Thoughts on solutions**

- **Potential approach**
   - Have MAS continue to run the logic to identify if a Private DBQ has been submitted with the claim or within 7 days of claim submission, before notifying VRO of automation eligible claims
   - For claims where a Private DBQ is found within 7 days of claim submission, have MAS notify VRO of the claim, including in the payload for that claim an indicator that a Private DBQ was found (e.g. add a boolean field to the endpoint for `privateDBQWithin7Days`)
   - When receiving a claim where (e.g.) `privateDBQWithin7Days`=`True` VRO incorporates new logic to mark RFD, generate an ARSD, and add a VBMS note to that claim

**Comment by dianagriffin** [comment]
This might be the same as the 4142 notice in #1308 -- need to verify

**Comment by shannonkford** [comment]
@dianagriffin I would be surprised if this is the same as 1308 - a 4142 is the Veteran giving permission to the VA to contact a doctor to get their private medical records; a private DBQ is a C&P exam that the Veteran asked their own doctor to do (outside of the VA exam system), who then fills out the DBQ which the Veteran then submits with their claim. 

I have many questions about this, including can we assume that the correct DBQ has been filled out? We've heard claims processors say that private DBQs are sometimes wrong or haven't asked for the right opinion or don't provide rationale for the nexus. I've also seen them call into question those that come from a "dbq mill" type doctor or practice.  

Also is this 7 days before the claim was submitted? 7 days after? Does the MAS system go back to the claim and check 7 days after?  I could see this as a distinct subset of RRD that requires different checks and balances and therefore isn't part of feature parity for the core RRD product.

Also curious what identifies something as a private DBQ versus a DBQ submitted by a contract examiner (maybe how doc type is set or something else at CEST?)

**Comment by dianagriffin** [comment]
@shannonkford you are right! I went digging in the MAS docs (section 5.9.7 [here](https://dvagov.sharepoint.com/:w:/r/sites/vaabdvro/Shared%20Documents/VRO%20-%20Virtual%20Regional%20Office/VRO%20Integrations/MAS%20Integrations/VRO%20Integration%20-%20APCAS%20Standardized%20Automation%20(March%202023).docx?d=w6780247c75fd48f898ad4ff60959fddb&csf=1&web=1&e=JghK6F)) and confirmed this is _not_ the same as the 4142, they just have similar logic requirements for the two scenarios are similar (e.g "if there is X in the last 7 days, do Y"). 

In summary, if a Private DBQ is found in the eFolder in the 7 days prior to claim submission (that's my interpretation of "within 7 days", it's actually not clearly defined in the docs), MAS automatically sets it RFD; same if a Private DBQ is submitted with the claim.
